hledger/Hledger/Cli/Utils.hs

```diff
@@ -117,6 +118,31 @@ journalApplyValue ropts j = do
                 = id
     return $ convert j
 
+-- | Run PeriodicTransactions from journal from today or journal end to requested end day.
+-- Add generated transactions to the journal
+journalAddForecast :: CliOpts -> Journal -> IO Journal
+journalAddForecast opts j = do
+  today <- getCurrentDay
+  let startDate =
+        -- Create forecast starting from today or end of journal, whatever is earliest, and until the end of requested reporting period
+        -- If end is not provided, do 180 days of forecast
+        case spanEnd (jdatespan j) of
+          Just jEnd -> min jEnd today
+          Nothing -> today
+      endDate = fromMaybe (addDays 180 today) $ periodEnd (period_ ropts)
+      dates = DateSpan (Just startDate) (Just endDate)
+      withForecast = [makeForecast t | pt <- jperiodictxns j, t <- runPeriodicTransaction pt dates] ++ (jtxns j)
```

Comment:
> Regarding `--forecast`: i think that this switch should "expand" periodic transaction, and automatic transactions might be triggered in addition with `--auto`. Do you envision problems/limitations here?

I see. I thought that you were going to expand on a whole period of journal and thus overlap with real expenses. That's why I suggested to drop real transactions from journal and show only generated from periodic transactions, but for `balance` report it is really useful to have them.

Potential problem here is that you may get forecast overlapping with real expenses if your last transaction/today is not at the end of budget period.

```
2016/12/20
    expenses:housing:rent  $600
    assets:cash

~ monthly from 2016/1/25
    income  $-1000
    expenses:housing  $700
    assets
```

As a forecast based on budget I would expect to see $600 under `expenses:housing:rent` and $100 under `expenses:housing`.
But if you'll consider "periodic transactions" as independent of budgeting - then it is totally fine.

Reply:
You have a point. It is best to start forecast immediately after the end of the journal (on the next day). Updating pr

Reply:
I actually didn't thought about that. I referred to overlap happending during reporting. If you use  `--forecast -M` and example I provided you'll get under category `expenses:housing` amount of`$1300` for `2016/12` because from periodic transaction we'll generate same `$700` amount though in a same budget period we already paid `$600`.

```
           || 2016/12 
===========++=========
 assets    ||   $-300 
   cash    ||   $-600 
 expenses  ||   $1300 
   housing ||   $1300 
     rent  ||    $600 
 income    ||  $-1000 
-----------++---------
           ||       0 
```

It feels like for budget-based forecasting it is a bit wrong and you should just expect that by the end of `2016/12` you'll spend `$700` (extra `$100` on top of real posting `$600`).

But "budget period" that's a notion that we are missing right now. I don't think therer is any actions at this moment regarding this issue.